### PR TITLE
feat: fix cancel button

### PR DESCRIPTION
**Pull request by infonl-marcel**
fix cancel button and some more readable switch statements for form fields handling

Solves: PZ-4224

### CODE REVIEW

File: src/main/app/src/app/informatie-objecten/informatie-object-add/informatie-object-add.component.ts [hunk 1]
```diff
@@ -333,23 +333,29 @@ export class InformatieObjectAddComponent implements AfterViewInit, OnDestroy {
...
-        } else if (key === "bestand") {
-          infoObject["bestandsomvang"] = value.size;
-          infoObject["bestandsnaam"] = value.name;
-          infoObject["bestand"] = value;
-          infoObject["formaat"] = value.type;
-        } else {
-          infoObject[key] = value;
+
+        switch (key) {
+          case "informatieobjectTypeUUID":
```
Comment (edgarvonk) [inline src/main/app/src/app/informatie-objecten/informatie-object-add/informatie-object-add.component.ts:338]: Keys like `informatieobjectTypeUUID` are also defined in `zac-openapi-types.d.ts` (which is automatically generated from the backend API). Is it not possible to use those here I wonder?

Comment (infonl-marcel) [reply src/main/app/src/app/informatie-objecten/informatie-object-add/informatie-object-add.component.ts:338]: Would be great if possible! I talked with Bas about this and just mentioned this improvement to Karin. We should use some generated type safety indeed. I am curious in what is exactly at hand and we can use already.

Comment (edgarvonk) [reply src/main/app/src/app/informatie-objecten/informatie-object-add/informatie-object-add.component.ts:338]: We started generating this code to gain more type safety indeed (Rick initially set this up). As far as I know we use it in several places already.

Comment (infonl-marcel) [reply src/main/app/src/app/informatie-objecten/informatie-object-add/informatie-object-add.component.ts:338]: But not open-api'd in this task for sure! @edgarvonk

Comment (edgarvonk) [reply src/main/app/src/app/informatie-objecten/informatie-object-add/informatie-object-add.component.ts:338]: I'm not sure exactly how those OpenAPI types are used in the frontend code. Would be good to delve into that. :-)

**Review by edgarvonk — APPROVED** [review]

File: src/main/app/src/app/informatie-objecten/informatie-object-add/informatie-object-add.component.ts (hunk 1 above)
Comment (infonl-marcel) [reply src/main/app/src/app/informatie-objecten/informatie-object-add/informatie-object-add.component.ts:338]: In docs/development/endToEndTypeSafety.md I read:

_**Good TypeScript code is written with as little TypeScript code as possible.**_

I need to take a moment to digest 😉

**Pull request merged**